packages/interface/src/components/complementary-area-toggle/index.js

```diff
@@ -32,18 +32,21 @@ function ComplementaryAreaToggle( {
 	const { enableComplementaryArea, disableComplementaryArea } = useDispatch(
 		'core/interface'
 	);
+	const componentProps = { ...props };
+	if ( ComponentToUse !== Button ) {
+		componentProps.isSelected = isSelected;
+	}
 	return (
 		<ComponentToUse
 			icon={ selectedIcon && isSelected ? selectedIcon : icon }
-			isSelected={ isSelected }
```

Comment:
Maybe we should just remove it? Is it useful anywhere?

If I understand properly the components that can be used as `as` should follow the same API as buttons.

Reply:
it's better that way. I'll do it.

Reply:
@youknowriad fixed 👍